cpp/modmesh/buffer/ConcreteBuffer.cpp

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2024, An-Chi Liu <phy.tiger@gmail.com>
```

Comment:
When copying and paste, please retain the original copyright tag.  It is required by the license pasted here.